File: rb/features.rb
# CloseApproachData SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module CloseApproachDataFeatures
  def self.make_feature(name)
    case name
    when "base"
      CloseApproachDataBaseFeature.new
    when "test"
      CloseApproachDataTestFeature.new
    else
      CloseApproachDataBaseFeature.new
    end
  end
end

File: py/features.py
# CloseApproachData SDK feature factory

from feature.base_feature import CloseApproachDataBaseFeature
from feature.test_feature import CloseApproachDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CloseApproachDataBaseFeature(),
        "test": lambda: CloseApproachDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
